sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/FileWriteResult.java

```diff
@@ -17,197 +17,69 @@
-      Map<Types.StructType, String> dataFileNames =
-          ImmutableMap.of(
-              dataFileStruct, "org.apache.iceberg.GenericDataFile",
-              partitionType, "org.apache.iceberg.PartitionData");
-      return AvroSchemaUtil.convert(dataFileStruct, dataFileNames);
-    }
-
-    @Override
-    public void encode(FileWriteResult value, OutputStream outStream)
-        throws CoderException, IOException {
-      // "version" of this coder.
-      // If breaking changes are introduced (e.g. from Beam, Iceberg, Avro, etc..),
-      // then update this version and create a fork in decode() below for the new decode logic.
-      // This helps keep the pipeline update-compatible
-      outStream.write(VERSION);
-
-      tableIdentifierCoder.encode(value.getTableIdentifier().toString(), outStream);
-      partitionSpecCoder.encode(value.getPartitionSpec(), outStream);
-      dataFileBytesCoder.encode(
-          AvroEncoderUtil.encode(value.getDataFile(), getDataFileAvroSchema(value)), outStream);
-    }
-
-    @Override
-    public FileWriteResult decode(InputStream inStream) throws CoderException, IOException {
-      // Forking logic can be added here depending on the version of this coder
-      assert inStream.read() == 0;
-
-      TableIdentifier tableId = TableIdentifier.parse(tableIdentifierCoder.decode(inStream));
-      PartitionSpec partitionSpec = partitionSpecCoder.decode(inStream);
-      DataFile dataFile =
-          checkArgumentNotNull(
-              AvroEncoderUtil.decode(dataFileBytesCoder.decode(inStream)),
-              "Decoding of dataFile resulted in null");
-      return FileWriteResult.builder()
-          .setTableIdentifier(tableId)
-          .setDataFile(dataFile)
-          .setPartitionSpec(partitionSpec)
-          .build();
-    }
-
-    @Override
-    public List<? extends Coder<?>> getCoderArguments() {
-      return Collections.emptyList();
-    }
-
-    @Override
-    public Object structuralValue(FileWriteResult fileWriteResult) {
-      return new FileWriteResultDeepEqualityWrapper(fileWriteResult);
-    }
-
-    @Override
-    public void verifyDeterministic() throws NonDeterministicException {}
-
-    @Override
-    public TypeDescriptor<FileWriteResult> getEncodedTypeDescriptor() {
-      return TypeDescriptor.of(FileWriteResult.class);
+  @SchemaIgnore
+  public TableIdentifier getTableIdentifier() {
+    if (cachedTableIdentifier == null) {
+      cachedTableIdentifier = TableIdentifier.parse(getTableIdentifierString());
     }
+    return cachedTableIdentifier;
+  }
 
-    public static FileWriteResultCoder of() {
-      return SINGLETON;
+  @SchemaIgnore
+  public ManifestFile getManifestFile() {
+    if (cachedManifestFile == null) {
+      try {
+        cachedManifestFile = ManifestFiles.decode(getManifestFileBytes());
+      } catch (IOException exc) {
+        throw new RuntimeException("Error decoding manifest file bytes");
+      }
```

Comment:
Can we include the table identifier in this error message for debuggability? or is it PII?

Reply:
PII

Reply:
We might be able to include the manifest file name. I think with the LGTM we have it would be worthwhile to merge so it is safe to release and then improve error message if we have time.